### DISCUSSION TITLE
Congratulations

### DISCUSSION THREAD

**Discussion by JPGuillemin** [z-huang/music#255] [General]
I may have tested hundreds of music apps , including everything related to YTM.... and I just wanted to tell you that I'm falling in love with Music !!!
Please add the ability to add an Artist to collection, and... to remove Artist from collection.
I also would like to restrict Home suggestions to styles related to my collection.
Fantastic app!!

**Comment by z-huang** [comment]
> Add/remove an artist to collection

Can you specify more? Currently, the artist tab shows all artists contained in your songs and albums.

**Reply by JPGuillemin** [reply]
I mean the same as adding an album to the collection : currently I can right click on an Album -> Add to Collection. 
When my Search result show an Artist channel I want to add : I would right click on the Artist channel to add it to the Artists tab of the collection.

**Reply by JPGuillemin** [reply]
Also : at the moment the only way to remove an Artist from the Artists tab is to search for the song where this artist is mentioned in tags, and to remove the song from collection.... Theorically : I have tried this way and it doesn't seems to work.

**Reply by z-huang** [reply]
If you want to add an artist to library, it means you like his/her songs. So, just add the artist's song to library and the artist will show up in the artist tab. This is my idea. Is this suitable for you, or do you have other idea?

> I have tried this way and it doesn't seems to work.

Does it work in the latest debug apk? (There is a bug when deleting albums in 0.4.1.)

**Reply by JPGuillemin** [reply]
> just add the artist's song to library and the artist will show (...) This is my idea. Is this suitable for you, or do you have other idea?

It "would" be OK for me theoretically , but in practice it's inconvenient :
1. If the song has many contributing Artists : all of them will be added, the collection will get cluttered with many unwanted Artists
2. There's no way for the moment to remove an Artist from Collection (while keeping some songs or albums where he/she's mentioned in tags in the Collection)

For example : I added a David Crosby album to the collection, where Madonna has contributed to one of the songs of the album : Madonna gets added to Artists Tab !! (Artists Tab  is my main startup tab) : I want to keep David Crosby, I want to keep the album : but I don't want Madonna to show in my main startup Artists Tab :)

Below too ways I would like to do it : 
1) Launch a search -> The search returns an Artist -> Long Press on Artist -> Add to Collection
2) I play a song -> Long Press on the Song's Artist -> Add to Collection

> Does it work in the latest debug apk?

Please point me to the download link for the built ARM64 APK (no way to easily build APKs here, I'm no Android dev ;) )

Daily master snapshots builds would be great :) I would beta test everything you want ;)

All the best

**Reply by z-huang** [reply]
For debug apk download link, see README.
The possible solution is to allow user to "hide" artists. But I'm not sure I'll do this.

**Reply by JPGuillemin** [reply]
OK, I figured out how to download the latest debug APK : I confirm that the collection cleanup problem is fixed : removing an album removes the associated Artist (at least if it's the only Artist's album).
On the other side : the player seek bar problem isn't fixed : still stuck at 100%.

Are the debug APK sending some telemetry ? do they eat more CPU?

**Reply by JPGuillemin** [reply]
about the problem of unwanted Artists : would it be possible to use ONLY the main album/song artist to trigger adding the Artist to collection ? (ignoring secondary contributors artists) ?

**Reply by z-huang** [reply]
> Are the debug APK sending some telemetry ? do they eat more CPU?

1. No. 2. Not sure.
Debug APK isn't that obfuscated and optimized as release APK. So, it may be slower.

> would it be possible to use ONLY the main album/song artist to trigger adding the Artist to collection ?

In current database design, no. All artists are stored in the database, so all artists will show up in the artist tab.
